zfs/replicate/snapshot/type.py

```diff
@@ -1,19 +1,21 @@
 # -*- coding: utf-8 -*-
 """ZFS Snapshot Type."""
-from typing import Any, NamedTuple, Optional
+from dataclasses import dataclass
+from typing import Optional
 
 from ..filesystem import FileSystem
 
 
-class Snapshot(NamedTuple):
+@dataclass
+class Snapshot:
     """ZFS Snapshot Type."""
 
     filesystem: FileSystem
     name: str
-    previous: Optional[Any]  # mypy doesn't support nested types yet
+    previous: Optional["Snapshot"]  # mypy doesn't support nested types yet
     timestamp: int
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
```

Comment:
Would it make sense for other to be `: "Snapshot"` too?

Reply:
mypy or pep8 wants the inherited signature to match so object is the only correct answer here.